File: engine/engine-model/src/main/scala/tools/packet/rps/RPSSelection.scala
package tools.packet.rps

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class RPSSelection(private var _selection: Byte, private var _answer: Byte) extends PacketInput {
  def selection: Byte = _selection

  def answer: Byte = _answer

  override def opcode(): SendOpcode = SendOpcode.RPS_GAME
}

File: engine/engine-model/src/main/java/tools/packet/rps/RPSSelection.java
package tools.packet.rps;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record RPSSelection(Byte selection, Byte answer) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.RPS_GAME;
   }
}